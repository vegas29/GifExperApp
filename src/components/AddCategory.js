import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        console.log('HandleSubmit', inputValue);

        if(inputValue.trim().length >2){
            setCategories( cats => [inputValue, ...cats ]);
            setInputValue('');
        }

        
    }

    return (
        <form
                onSubmit={handleSubmit}
            >
                <input
                    type="text"
                    onChange={handleInputChange}
                    value={inputValue}
                    className="mt-5 rounded-xl bg-gray-100 focus:outline-none focus:border-2 focs:border-gray-200 py-2 px-5"
                    placeholder="Digite el nombre del gif a buscar"
                />
        </form>
    );

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


 
export default AddCategory;