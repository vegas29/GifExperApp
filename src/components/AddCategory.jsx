import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target})=>{
        setInputValue(target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length <= 1) return;

        setInputValue('');
        onNewCategory( inputValue.trim() );
        
    }

    return (
        <form
            onSubmit={handleSubmit}
            aria-label="form"
        >
            <input
                type="text"
                onChange={ handleInputChange }
                value={ inputValue }
                className="mt-5 rounded-xl bg-gray-100 focus:outline-none focus:border-2 focs:border-gray-200 py-2 px-5 text-xl"
                placeholder="Digite el nombre del gif a buscar"
            />
        </form>
    );

}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}

