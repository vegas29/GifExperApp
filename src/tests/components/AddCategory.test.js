import react from 'react';
import {shallow} from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory', ()=>{

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    test('Debe de mostrarse correctamente', ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de cambiar la caja de texto', ()=>{
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change', {
            target:{
                value: value
            }
        });

        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('No debe de postear la infomraciÃ³n on submit', ()=>{
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('Deben de llamar el setCategories y limpiar la caja de texto', ()=>{
        const value = 'hola mundo';
        wrapper.find('input').simulate('change', { target: {value}});

        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));
        //expect(wrapper.find('input').text()).toBe('');
        expect(wrapper.find('input').prop('value')).toBe('');
    })
})