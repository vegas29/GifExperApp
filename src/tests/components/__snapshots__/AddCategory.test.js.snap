// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="mt-5 rounded-xl bg-gray-100 focus:outline-none focus:border-2 focs:border-gray-200 py-2 px-5"
    onChange={[Function]}
    placeholder="Digite el nombre del gif a buscar"
    type="text"
    value=""
  />
</form>
`;
